### PR TITLE
Backlog 16049 table

### PR DESCRIPTION
**Pull request by AKarmanov**
## JIRA

https://jira.jahia.org/browse/BACKLOG-16049

## Description

First level moonstone table integration. Look at readme in ContentListTable for deployment instructions. Will work with latest moonstone which will incorporate branch MOON-297.

### CODE REVIEW

**Review by gflores-jahia — COMMENTED** [review]
lgtm for the most part and was really well done. I noticed that we're missing the data-cm-role tags in some of the cells/row. Wondering if we'd need to add that here.

File: src/javascript/JContent/ContentRoute/ContentLayout/ContentListTable/ContentListTableMoon.jsx [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+        setPath(siteKey, node.path, newMode, {sub: node.primaryNodeType.name !== 'jnt:page' && node.primaryNodeType.name !== 'jnt:contentFolder'});
+    };
+
+    let columnData = previewState === CM_DRAWER_STATES.SHOW ? reducedColumnData : allColumnData;
+    let isPreviewOpened = previewState === CM_DRAWER_STATES.SHOW;
+
+    if (contentNotFound) {
+        return (
+            <ContentListTableWrapper>
+                <ContentNotFound columnSpan={columnData.length} t={t}/>
```
[the remark this answers is not in the document]
Comment (AKarmanov) [reply src/javascript/JContent/ContentRoute/ContentLayout/ContentListTable/ContentListTableMoon.jsx:120]: Yes, good catch. The original didn't but the moonstone version does.

File: src/main/resources/javascript/locales/en.json [hunk 2]
```diff
@@ -163,7 +163,8 @@
         "createdBy": "Created By",
         "name": "Name",
         "lastModified": "Last modified on",
-        "type": "Type"
+        "type": "Content",
```
[the remark this answers is not in the document]
Comment (AKarmanov) [reply src/main/resources/javascript/locales/en.json:166]: Indeed it is.

File: src/javascript/JContent/ContentRoute/ContentLayout/ContentListTable/todo.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+* remove table switcher (to switch use {type: 'tableType', payload: 'moon'})
```
[the remark this answers is not in the document]
Comment (AKarmanov) [reply src/javascript/JContent/ContentRoute/ContentLayout/ContentListTable/todo.md:1]: The plan was to remove it once we get rid of the original table. We should be able to lose it in the next iteration.

**Comment by AKarmanov** [comment]
>I noticed that we're missing the data-cm-role tags in some of the cells/row. Wondering if we'd need to add that here.

Oh, yes we should add it where we can. I'll take a look.

File: src/main/resources/javascript/locales/en.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (gflores-jahia) [reply src/main/resources/javascript/locales/en.json:166]: I see `type` was added for fr/de, but not `status`

**Review by gflores-jahia — APPROVED** [review]

**Pull request merged**